### DISCUSSION TITLE
snesCIC and DIY arduino pic programmer - Any luck anyone?

### DISCUSSION THREAD

**Discussion by Fabs-tinkers** [sanni/cartreader#921] [Need help]
Has anyone built and successfully programmed the device through the linked ( https://diyodemag.com/projects/arduino_pic_programmer ) guide in sanny wiki. Im asking since Ive done already 4-th breadboard build and something doesnt add up. Also seems to me that the schematics, photos and guide differ from each other. Also are there any other arduinobased pick programmers that you can follow step by steb and it actually works?

**Comment by Ancyker** [comment]
I had no luck with these and ended up buying a programmer:

![2023-05-04 18 26 33](https://github.com/sanni/cartreader/assets/7105333/061fa5e9-add0-4798-ad6c-d02c4c2012b5)

I eventually got an even better programmer as I started flashing more PICs:

![2023-12-27 23 50 25](https://github.com/sanni/cartreader/assets/7105333/5bd2e049-6237-454d-a94c-b9ee286f3da6)

**Comment by clarion78** [comment]
This video may be helpful.
(Time 33:33)
https://www.youtube.com/watch?v=iLT99VNh_ic&t=2388s
This video uses only K150 and does not require Arduino.

In my case, I succeeded in writing by creating PICerFT.
http://einstlab.web.fc2.com/PICerFT/PICerFT.html
p0ojisan has released the gerber file of PICerFT.
https://note.com/ojigaming/n/n42a4d90dbfee

I hope this helps :)

**Comment by Chomemel** [comment]
If someone make it work (flashing PICs via arduino), does it mean we could do that with an OSCR (with the appropriate code and the appropriate adapter) ?

**Reply by clarion78** [reply]
Maybe it's possible.
The Arduino pin number printed on the six slot adapter board seems to be a clue.

**Comment by clarion78** [comment]
I tried DIYODE's Arduino-based PIC programmer.
Writing has completed and OSCR has created a dump of the SA1 chip cartridge.
I had 4N35, so I used it instead of 4N25. ( I ordered 4N25 and will try it later. )
The wiring in the DIYODE circuit diagram and the breadboard image were different, so I wired it according to the circuit diagram.
![pic12f629_writer_arduino](https://github.com/sanni/cartreader/assets/124872141/be98fcd1-ecb5-4f35-813c-6e9ccc2d0034)

**Reply by Chomemel** [reply]
Congratulation ! That's excellent news.

You flashed the PIC with the CIC lock. Then solder it to the OSCR and then you were able to dump a SA-1 game. Did I get it right?

Maybe, one day, after I finish my million projects, I'll try to create a board to flash a PIC with the Flashrom adapter. But I won't be able to insert the code as I know zero about coding.

**Reply by clarion78** [reply]
thanks Chomemel :)
Since I had already assembled the cart reader v5 (HW5), I soldered the PIC to the [PIC ADAPTER](https://github.com/sanni/cartreader/blob/master/hardware/cic_adapter/PIC_adapter_gerber.zip).
I've incorporated it into OSCR and it seems to work fine.
![pic12f629_writer_arduino-2](https://github.com/sanni/cartreader/assets/124872141/2277cb48-5721-40ed-bc0a-2633b203f125)
![pic12f629_writer_arduino-1](https://github.com/sanni/cartreader/assets/124872141/d9786e71-57f2-4993-b853-f57c3c86d8db)
![pic12f629_writer_arduino-3](https://github.com/sanni/cartreader/assets/124872141/73a1527d-d521-4e4a-b58e-053fcb998fc0)

**Reply by Chomemel** [reply]
I realised I didn't really undertand the article. Maybe it's because I didn't really read the whole thing. I thought it was about creating a PIC programmer on its own without using a software on the computer. Like the flasrom programmer on the OSCR. So I thought it was possible to add a PIC programmer functionnality to the OSCR. But I got it wrong.
Anyway, nice work, [clarion78](https://github.com/clarion78).

**Reply by clarion78** [reply]
I recieved 4N25.
I replaced from 4N35 and it worked fine.
